02-household-queries/ingest.py

```diff
@@ -1,33 +1,107 @@
+import os
+import dotenv
+import json
 from langchain_community.document_loaders import PDFMinerLoader
 from langchain.docstore.document import Document
-from langchain_text_splitters import RecursiveCharacterTextSplitter
-import json
+from langchain_text_splitters import (
+    RecursiveCharacterTextSplitter,
+    NLTKTextSplitter,
+    SpacyTextSplitter,
+)
+
+from langchain_community.embeddings import (
+    SentenceTransformerEmbeddings,
+    HuggingFaceEmbeddings,
+)
+# from langchain_google_genai import ChatGoogleGenerativeAI, GoogleGenerativeAIEmbeddings
+
+from llm import ollama_client
+
+
+dotenv.load_dotenv()
+
+_llm_model_name = os.environ.get("LLM_MODEL_NAME", "mistral")
+
+llm = ollama_client(_llm_model_name, settings={"temperature": 0.1})
+
+
+EMBEDDINGS = {
+    "all-MiniLM-L6-v2": {
+        "func": SentenceTransformerEmbeddings(model_name="all-MiniLM-L6-v2"),
+        "token_limit": 256,
+    },
+    "HuggingFace::all-MiniLM-L6-v2": {
+        "func": HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2"),
+        "token_limit": 256,
+    },
+    # "Google::models/embedding-001": {
+    #     "func": GoogleGenerativeAIEmbeddings(model="models/embedding-001"),
+    #     "token_limit": 2048,
+    # },
+    # "google_models/text-embedding-004": {
+    #     "func": GoogleGenerativeAIEmbeddings(model="models/text-embedding-004"),
+    #     "token_limit": 768,
+    # },
+    "BAAI/bge-small-en-v1.5": {
+        "func": SentenceTransformerEmbeddings(model_name="BAAI/bge-small-en-v1.5"),
+        "token_limit": 512,
+    },
+    "mixedbread-ai/mxbai-embed-large-v1": {
+        "func": SentenceTransformerEmbeddings(
+            model_name="mixedbread-ai/mxbai-embed-large-v1"
+        ),
+        "token_limit": 1024,
+    },
+}
 
 
 # split text into chunks
 def get_text_chunks_langchain(
-    text, source, silent=False, chunk_size=750, chunk_overlap=100
+    text, source, chunk_size, chunk_overlap, token_limit, text_splitter_choice, silent
 ):
-    text_splitter = RecursiveCharacterTextSplitter(
-        chunk_size=chunk_size, chunk_overlap=chunk_overlap
-    )
+    if text_splitter_choice == "2":
+        text_splitter = NLTKTextSplitter()
+    elif text_splitter_choice == "3":
+        text_splitter = SpacyTextSplitter()
+    else:
+        text_splitter = RecursiveCharacterTextSplitter(
+            chunk_size=chunk_size, chunk_overlap=chunk_overlap
+        )
+
     entire_text = source + "\n\n" + text
     texts = text_splitter.split_text(entire_text)
+
     if not silent:
         print("  Split into", len(texts))
+    for t in texts:
+        token_count = llm.get_num_tokens(t)
+        overlap_and_token = token_count + chunk_overlap
+        if token_count > token_limit:
+            print(f"Exceeded token limit of {token_limit}: {token_count};")
+        elif chunk_size > (overlap_and_token):
+            print(
+                f"Exceeded token count and overlap {overlap_and_token}: {chunk_size};"
+            )
```

Comment:
- `chunk_size` and `chunk_overlap` are measured in number of characters
- `token_count` is measured in number of tokens/floats
What is the unit of measure for `overlap_and_token`? The number of tokens (for `token_count`) is not comparable to the number of characters (for `chunk_overlap`).

What is the reasoning behind this `elif` block?

Reply:
I was under the assumption the what was in the processed would be determined by the chunk size, overlap and the base token amount given the text. I'll remove this